activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -436,11 +436,30 @@ def advisory_locks_enabled? # :nodoc:
         supports_advisory_locks? && @advisory_locks_enabled
       end
 
+      # Obtains an exclusive session level advisory lock, if available, for the duration of the block.
+      #
+      # Returns +false+ without executing the block if the lock could not be obtained.
+      def with_advisory_lock(lock_id, timeout = 0)
+        lock_acquired = get_advisory_lock(lock_id, timeout)
+
+        if lock_acquired
+          yield
+
+          lock_acquired
+        else
+          lock_acquired
+        end
```

Comment:
How about to consolidate returning `lock_acquired` in if and else branches? It is just a preference though.

```suggestion
        if lock_acquired
          yield
        end

        lock_acquired
```